src/libraries/Microsoft.PowerFx.Interpreter/Functions/IRandomService.cs

```diff
@@ -31,12 +39,25 @@ internal class DefaultRandomService : IRandomService
         [ThreadSafeProtectedByLock(nameof(_randomizerLock))]
         private static readonly Random _random = new Random();
 
+        private static readonly object _randomizerDecimalLock = new object();
+
+        [ThreadSafeProtectedByLock(nameof(_randomizerDecimalLock))]
+        private static readonly Random _randomDecimal = new Random();
+
         public virtual double NextDouble()
         {
             lock (_randomizerLock)
             {
                 return _random.NextDouble();
             }
         }
+
+        public virtual decimal NextDecimal()
+        {
+            lock (_randomizerDecimalLock)
+            {
+                return new decimal(_randomDecimal.Next(), _randomDecimal.Next(), _randomDecimal.Next(542101087), false, 28);
```

Comment:
>return new decimal(_randomDecimal.Next(), _randomDecimal.Next(), _randomDecimal.Next(542101087), false, 28); [](http://example.com/codeflow?start=16&length=108)

It's possible that this will return a number greater than 1:

    var dec = new Decimal(
        int.MaxValue,
        int.MaxValue,
        542101086, false, 28);

returns

    1.0000000004745384014313947135

Reply:
i run manytime and always get less than 1 so I thought it was right. But sounds it can be more than 1. 
I updated logic here to make sure we get value less than 1.